Neos.Neos/Tests/Behavior/Features/Bootstrap/DispatcherTrait.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Neos.ContentRepository package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Behat\Gherkin\Node\PyStringNode;
+use GuzzleHttp\Psr7\Message;
+use Neos\ContentRepository\TestSuite\Behavior\Features\Bootstrap\CRTestSuiteRuntimeVariables;
+use Neos\Fusion\Core\Cache\ContentCache;
+use Neos\Neos\Domain\Service\FusionService;
+use PHPUnit\Framework\Assert;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestFactoryInterface;
+
+/**
+ * @internal only for behat tests within the Neos.Neos package
+ */
+trait DispatcherTrait
+{
+    use CRTestSuiteRuntimeVariables;
+
+    private ResponseInterface|null $response = null;
+
+    /**
+     * @template T of object
+     * @param class-string<T> $className
+     *
+     * @return T
+     */
+    abstract private function getObject(string $className): object;
+
+    /**
+     * @BeforeScenario
+     */
+    public function setupDispatcherTest(): void
+    {
+        $this->getObject(ContentCache::class)->flush();
+        $this->response = null;
+    }
+
+    /**
+     * @When the sites Fusion code is:
+     */
+    public function iHaveTheFollowingFusionCodeForTheSite(PyStringNode $fusionCode)
```

Comment:
😭 😭 the new `unsafeSetAdditionalFusionSourceCodeToThisSingleton` is exactly what we NOT want imo...